### PR TITLE
Improve error logging when checking the condition endpoint

### PR DESCRIPTION
**Pull request by Ahmed-Ghanam**
## Description
Add more details to the application insights when calling the condition endpoint.

## Related Issue(s)
- #801 

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [x] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [x] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

## Summary by CodeRabbit

- **Bug Fixes**
	- Improved error handling and robustness for notification sending conditions, ensuring more reliable processing in various failure scenarios.
- **New Features**
	- Enhanced logging for both successful and failed notification condition checks, providing clearer and more detailed diagnostic information.
- **Tests**
	- Expanded test coverage for edge cases and error handling in notification condition checks, ensuring greater reliability.

### CODE REVIEW

File: src/Altinn.Notifications.Integrations/SendCondition/SendConditionClient.cs [hunk 1]
```diff
@@ -24,31 +24,98 @@ public SendConditionClient(HttpClient client)
         /// <inheritdoc/>
         public async Task<Result<bool, ConditionClientError>> CheckSendCondition(Uri url)
         {
-            HttpResponseMessage res = await _client.GetAsync(url);
-            string responseString = await res.Content.ReadAsStringAsync();
+            try
+            {
+                using var response = await _client.GetAsync(url).ConfigureAwait(false);
```
[the remark this answers is not in the document]
Comment (Ahmed-Ghanam) [reply src/Altinn.Notifications.Integrations/SendCondition/SendConditionClient.cs:29]: It helps prevent potential deadlocks that can occur when synchronously waiting on asynchronous code that tries to get back to an already blocked context, and it avoids unnecessary thread switching.

[the remark this answers is not in the document]
Comment (Ahmed-Ghanam) [reply src/Altinn.Notifications.Integrations/SendCondition/SendConditionClient.cs:29]: In classic ASP .NET (the .NET Framework `System.Web` pipeline) each request installs an `SynchronizationContext` so that after an await the continuation “re‑enters” the request context (preserving things like `HttpContext.Current`, culture, identity, etc.).
ASP .NET Core never installs a `SynchronizationContext` for MVC/WebAPI pipelines—`SynchronizationContext.Current` is always null—so continuations simply run on whatever thread‐pool thread happens to pick them up. Because there is no `SynchronizationContext` in ASP .NET Core, there’s nothing for `.ConfigureAwait(false)` to “turn off.”

**Review by SandGrainOne — APPROVED** [review]
Multiple of cases where we're throwing an exception in order to fail a test that can be replaced by Assert.Fail()

File: test/Altinn.Notifications.Tests/Notifications.Integrations/SendCondition/SendConditionClientTests.cs [hunk 2]
```diff
@@ -21,11 +25,12 @@ public class SendConditionClientTests
             PropertyNamingPolicy = JsonNamingPolicy.CamelCase
         };
 
+        private readonly DelegatingHandlerStub _messageHandler;
         private readonly SendConditionClient _sendConditionClient;
 
         public SendConditionClientTests()
         {
-            var messageHandler = new DelegatingHandlerStub(async (request, token) =>
+            _messageHandler = new DelegatingHandlerStub(async (request, token) =>
```
[the remark this answers is not in the document]
Comment (Ahmed-Ghanam) [reply test/Altinn.Notifications.Tests/Notifications.Integrations/SendCondition/SendConditionClientTests.cs]: Converted it back to a private member
